ssh/ssh.go

```diff
@@ -163,11 +140,18 @@ func GetExeAndArgs(osEnv config.Environment, gitEnv config.Environment, meta *SS
 	multiplexing = false
 	multiplexEnabled := gitEnv.Bool("lfs.ssh.automultiplex", true)
 	if variant == variantSSH && multiplexDesired && multiplexEnabled {
-		controlPath, err := getControlDir(osEnv)
-		if err == nil {
-			multiplexing = true
-			controlPath = filepath.Join(controlPath, "sock-%C")
-			args = append(args, "-oControlMaster=auto", fmt.Sprintf("-oControlPath=%s", controlPath))
+		multiplexing = true
+		controlMasterArg := "-oControlMaster=no"
+		controlPath = multiplexControlPath
+		if multiplexControlPath == "" {
+			controlMasterArg = "-oControlMaster=yes"
```

Comment:
How does this work on the Windows OpenSSH, which [doesn't support `ControlMaster`](https://github.com/PowerShell/Win32-OpenSSH/issues/1328)?  Is it at least no less functional?

Reply:
What happens right now if you pass `-oControlMaster=auto` to Windows OpenSSH? Does it just not do anything with it? If so, does `-oControlMaster=yes` or `-oControlMaster=no` also not do anything?

Reply:
Looking at that issue, it seems that the functionality is so broken that they haven't even implemented a "This doesn't work on Windows" error based on [this comment](https://github.com/PowerShell/Win32-OpenSSH/issues/1328#issuecomment-922003814):

> Can we at least have the client report that the feature is not supported instead of printing opaque errors about sockets, as though the feature were implemented but happened to have suffered some incidental mishap?

Would this test be sufficient?

```go
if variant == variantSSH && runtime.GOOS != "windows" && multiplexDesired && multiplexEnabled {
```

Reply:
I don't think that's the right behaviour, since it apparently does work for Git for Windows's OpenSSH.  Perhaps the default behaviour of the setting above should be `runtime.GOOS != "windows"` instead of `true`.  That would at least allow users to override it if they're using the OpenSSH from Git for Windows.

Reply:
OK, I've implemented your suggestion.

Reply:
Actually, now I'm wondering if we should simply have Windows OpenSSH users opt out instead of requiring Git for Windows OpenSSH users to opt in. Seems to me like someone who's using Git is more likely to use Git for Windows OpenSSH. WDYT?

Reply:
Sure, that seems fine.  If the default behaviour doesn't cause any problems (as the message you quoted implies) and is just less functional, then it should be fine to leave it enabled by default.

Reply:
> If the default behaviour doesn't cause any problems (as the message you quoted implies)

Well, to me the message implies that `-oControlMaster=yes` on Windows produces cryptic errors, which means this behavior doesn't work. My thinking was that if someone's using Git on Windows, they're more likely to be using Git's OpenSSH, so we should cater to them by default and just hope they're not using Windows OpenSSH. However, if we want to make things always work by default, even for the Windows OpenSSH minority, then this should be off by default on Windows.

Reply:
I'm keeping the multiplexing off by default on Windows, and (I think) I've fixed the CI issues.